src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandRunner.cs

```diff
@@ -0,0 +1,548 @@
+            var msBuild = new MSBuildAPIUtility(whyCommandArgs.Logger);
+
+            string targetPackage = whyCommandArgs.Package;
+
+            IEnumerable<string> projectPaths = Path.GetExtension(whyCommandArgs.Path).Equals(".sln")
+                                                    ? MSBuildAPIUtility.GetProjectsFromSolution(whyCommandArgs.Path).Where(f => File.Exists(f))
+                                                    : new List<string>() { whyCommandArgs.Path };
+
+            foreach (var projectPath in projectPaths)
+            {
+                Project project = MSBuildAPIUtility.GetProject(projectPath);
+
+                // if the current project is not a PackageReference project, print an error message and continue to the next project
+                if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
+                {
+                    Console.Error.WriteLine(
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.Error_NotPRProject,
+                            projectPath));
+
+                    ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+                    continue;
+                }
+
+                string projectName = project.GetPropertyValue(ProjectName);
+                string assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+
+                // if the assets file was not found, print an error message and continue to the next project
+                if (!File.Exists(assetsPath))
+                {
+                    Console.Error.WriteLine(
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.Error_AssetsFileNotFound,
+                            projectPath));
+
+                    ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+                    continue;
+                }
+
+                var lockFileFormat = new LockFileFormat();
+                LockFile assetsFile = lockFileFormat.Read(assetsPath);
+
+                // assets file validation
+                if (assetsFile.PackageSpec == null
+                    || assetsFile.Targets == null
+                    || assetsFile.Targets.Count == 0)
+                {
+                    Console.Error.WriteLine(
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.WhyCommand_Error_CannotReadAssetsFile,
+                            assetsPath));
+
+                    ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+                    continue;
+                }
+
+                // get all resolved package references for a project
+                List<FrameworkPackages> packages = msBuild.GetResolvedVersions(project, whyCommandArgs.Frameworks, assetsFile, transitive: true);
+
+                if (packages?.Count > 0)
```

Comment:
Packages can't be null.
```suggestion
                if (packages.Count > 0)
```

Reply:
`msBuild.GetResolvedVersions` says it returns null on errors, so these 'packages' (`List<FrameworkPackages>`) may well be null here in some cases: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs#L661

Reply:
We're not using the `GetResolvedVersions()` method any longer, so the return type doesn't need a null check here anymore. We're just checking `.Count > 0` now.